lib/Utils.groovy

```diff
@@ -4,44 +4,548 @@
 
 import org.yaml.snakeyaml.Yaml
 
+import nextflow.Nextflow
+import nextflow.splitter.SplitterEx
+
 class Utils {
 
-    //
-    // When running with -profile conda, warn if channels have not been set-up appropriately
-    //
-    public static void checkCondaChannels(log) {
-        Yaml parser = new Yaml()
-        def channels = []
+    public static parseInput(input_fp_str, stub_run, log) {
```

Comment:
Not something breaking right now but have you considered using `nf-validation` to perform your parameter validation and your samplesheet validation and conversion to a channel? It would increase the maintainability of the pipeline a lot in the long run.

Reply:
Hi @nvnieuwk, thanks for the comment!

We've had some discussion on this earlier [here](https://github.com/nf-core/oncoanalyser/pull/29#discussion_r1609454595), and I think it makes sense to do. Since this isn't breaking I'd like to make this change in concert with [the migration to nf-schema](https://github.com/nf-core/oncoanalyser/issues/51) following the initial release.

Reply:
Awesome thank you for the answer! Looking forward to the implementation once nf-core fully supports it :)